challenges/lib/src/main/java/challenges/utilities/AnimalShelter.java

```diff
@@ -0,0 +1,31 @@
+package challenges.utilities;
+import challenges.stacksandqueues.Queue;
+
+public class AnimalShelter {
+    Queue<Cat> cats = new Queue<>();
+    Queue<Dog> dogs = new Queue<>();
+
+    public void enqueue(Dog dog){
+        dogs.enquque(dog);
+
+    }
+    public void enqueue(Cat cat){
+        cats.enqueue(cat);
+    }
+    public Animal dequeue(String animal) throws Exception{
+        if (dogs.isEmpty() && cats.isEmpty()){
+            throw new Exception("Fresh out of Pets");
+        }
+
+        if("dog".equals(animal) && !dogs.isEmpty()) return dogs.dequeue();
```

Comment:
Your logic is broken in the Queue implementation. You currently have a Queue throwing a NoSuchElementException if isempty() returns.